src/eth/rpc.ts

```diff
@@ -1,25 +1,42 @@
 import  {bondingManager, roundsManager, merkleSnapshot} from './contracts'
 import {BigNumber, utils} from 'ethers'
 const { createApolloFetch } = require('apollo-fetch');
+const provider = require('./provider')
 
 const fetchSubgraph = createApolloFetch({
     uri: `${process.env.SUBGRAPH_URL}`,
   });
 
 export const getEarnings = async(address:string, endRound: BigNumber, delegateAddr?: string): Promise<{delegator: string, delegate: string, pendingStake: BigNumber, pendingFees: BigNumber}> => {
+    const [pendingStake, pendingFees] = await Promise.all([
+        bondingManager.pendingStake(address, endRound, { gasLimit: BigNumber.from("1000000000000000000") }),
+        bondingManager.pendingFees(address, endRound, { gasLimit: BigNumber.from("1000000000000000000") })
+    ])
+    
     try {
         const earnings: {delegator: string, delegate: string, pendingStake: BigNumber, pendingFees: BigNumber} = {
             delegator: address,
             delegate: delegateAddr || "",
-            pendingStake: await bondingManager.pendingStake(address, endRound, { gasLimit: BigNumber.from("1000000000000000000") }),
-            pendingFees: await bondingManager.pendingFees(address, endRound, { gasLimit: BigNumber.from("1000000000000000000") })
+            pendingStake,
+            pendingFees
         }
         return earnings
     } catch(err) {
         return err
     }
 }
 
+const isEOA = async (address: string) => {
+    return provider.getCode(address) === "0x"
+}
+
+const filterAddresses = async (arr) => {
```

Comment:
Realizing that we can also filter out orchestrators (i.e. delegator == delegate) since only delegators are eligible for snapshot migrations. Feel free to address here or in a separate PR.